framework/OutStreams/GeneralPlot.py

```diff
@@ -146,6 +189,106 @@ def __init__(self):
+          self.options[subnode.getName()]['text'    ] = xmlNode.attrib['name']
+        if 'location' not in self.options[subnode.getName()].keys():
+          self.options[subnode.getName()]['location'] = 'center'
+      ## is this 'figureProperties' valid?
+      if subnode.getName() == 'figureProperties':
+        self.options[subnode.getName()] = {}
+        for subsub in subnode:
+          self.options[subnode.getName()][subsub.getName()] = subsub.text.strip()
+    self.type = 'OutStreamPlot'
+
+    if not 'plotSettings' in self.options.keys():
+      self.raiseAnError(IOError, 'For plot named ' + self.name + ' the plotSettings block is required.')
+
+    if not foundPlot:
+      self.raiseAnError(IOError, 'For plot named' + self.name + ', No plot section has been found in the plotSettings block!')
+
+    self.outStreamTypes = []
+    xyz, xy             = sorted(['x','y','z']), sorted(['x','y'])
+    for pltIndex in range(len(self.options['plotSettings']['plot'])):
+      if not 'type' in self.options['plotSettings']['plot'][pltIndex].keys():
+        self.raiseAnError(IOError, 'For plot named' + self.name + ', No plot type keyword has been found in the plotSettings/plot block!')
+      else:
+        # check the dimension and check the consistency
+        if set(xyz) < set(self.options['plotSettings']['plot'][pltIndex].keys()):
+          dim = 3
+        elif set(xy) < set(self.options['plotSettings']['plot'][pltIndex].keys()):
+          dim = 2 if self.options['plotSettings']['plot'][pltIndex]['type'] != 'histogram' else 3
+        elif set(['x']) < set(self.options['plotSettings']['plot'][pltIndex].keys()) and self.options['plotSettings']['plot'][pltIndex]['type'] == 'histogram':
+          dim = 2
+        else:
+          self.raiseAnError(IOError, 'Wrong dimensionality or axis specification for plot '+self.name+'.')
+        if self.dim is not None and self.dim != dim:
+          self.raiseAnError(IOError, 'The OutStream Plot '+self.name+' combines 2D and 3D plots. This is not supported!')
+        self.dim = dim
+        if self.availableOutStreamTypes[self.dim].count(self.options['plotSettings']['plot'][pltIndex]['type']) == 0:
+          self.raiseAMessage('For plot named' + self.name + ', type ' + self.options['plotSettings']['plot'][pltIndex]['type'] + ' is not among pre-defined plots! \n The OutstreamSystem will try to construct a call on the fly!', 'ExceptedError')
+        self.outStreamTypes.append(self.options['plotSettings']['plot'][pltIndex]['type'])
+    self.raiseADebug('matplotlib version is ' + str(matplotlib.__version__))
+
+    if self.dim not in [2, 3]:
+      self.raiseAnError(TypeError, 'This Plot interface is able to handle 2D-3D plot only')
+
+    if 'gridSpace' in self.options['plotSettings'].keys():
+      grid = list(map(int, self.options['plotSettings']['gridSpace'].split(' ')))
+      self.gridSpace = matplotlib.gridspec.GridSpec(grid[0], grid[1])
```

Comment:
since we do not use it, can we comment this out (for coverage purposes...we never know when we can finalize this :D and it might be quite a long time :D )

Reply:
This came directly from the original code, I think, so I didn't modify it. If you let me know how you want this changed, I can change this.

Reply:
nono I mean
the handle_future